src/layers/logging.rs

```diff
@@ -86,6 +94,140 @@ impl Layer for LoggingLayer {
     }
 }
 
+/// `LoggingReader` is a wrapper of `BytesReader`, with logging functionality.
+pub struct LoggingReader {
+    scheme: Scheme,
+    path: String,
+    op: Operation,
+    have_read: usize,
+    to_read: Option<u64>,
+    inner: BytesReader,
+}
+
+impl LoggingReader {
+    pub(crate) fn make_reader(
+        scheme: Scheme,
+        op: Operation,
+        path: &str,
+        size: Option<u64>,
+        reader: BytesReader,
+    ) -> BytesReader {
+        let r = Self {
+            scheme,
+            op,
+            path: path.to_string(),
+            have_read: 0,
+            to_read: size,
+            inner: reader,
+        };
+        Box::new(r)
+    }
+}
+
+impl AsyncRead for LoggingReader {
+    fn poll_read(
+        mut self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+        buf: &mut [u8],
+    ) -> Poll<Result<usize>> {
+        Pin::new(&mut (*self.inner)).poll_read(cx, buf).map(|r|{
+           match r {
+           Ok(r) => {
+               self.have_read += r;
+               debug!(target: "opendal::services", "service={} operation={} path={} size={:?} have_read={} poll read -> got: {}B", self.scheme, self.op, self.path, self.to_read, self.have_read, r);
```

Comment:
```suggestion
               trace!(
                   target: "opendal::services", "service={} operation={} path={} size={:?} has_read={} -> read {}B", self.scheme, self.op, self.path, self.to_read, self.have_read, r);
```

Too verbose. Please consider using `trace` instead.